src/connection.ts

```diff
@@ -1111,6 +1132,17 @@ export class Connection extends EventEmitter {
       }
 
       if (responsePacket.ilpPacketType === IlpPacketType.Fulfill) {
+        for (let frame of responsePacket.frames) {
+          if (frame.type === FrameType.StreamReceipt) {
```

Comment:
:thinking: I'm not sure
On the plus side, detecting an invalid receipt (by checking for consistent nonce and accurate id, amount & stream start time) could allow the sender to immediately stop sending money.
OTOH:
- the sender can't verify the hmac
- this STREAM implementation isn't currently tracking the total amount received by the counterparty for a stream (we're just taking their word for it:
https://github.com/interledgerjs/ilp-protocol-stream/blob/476c12daf215fbb7573174de1351f4cc1d78a13d/src/connection.ts#L733-L739

Reply:
On the connection level, if the packet is fulfilled, the sender can & should assume that at least the minimum destination amount from the Prepare got delivered to the recipient ([this line](https://github.com/interledgerjs/ilp-protocol-stream/blob/681cd740f5f619994c01f8922f734eedaba50b65/src/connection.ts#L1151) should probably be changed to `addTotalDelivered(max(requestPacket.prepareAmount, responsePacket.prepareAmount))`). Likewise, the amount of the receipt could be validated against the cumulative amount we know got delivered to the recipient.

_But_, because receipt accounting is on a per-stream basis rather than per-connection, it might be complicated to compute the amount delivered on each stream

Reply:
Is it a concern if the receiver underreports the amount they receive to the verifier? Or is it assumed that the verifier is trusting the receiver to honestly report the amount they receive?

Reply:
I think the sender is the only who stands to lose anything from an underreported receipt amount.
Unless there was some scenario where the verifier was confirming that a certain amount *hasn't* been paid, but that doesn't seem doable with depending on the sender to get receipts to the verifier.